### PR TITLE
Add custom reference field for manual capture 

### PR DESCRIPTION
**Pull request by leungkinghin**
- Add new custom attribute reference for manual capture requests
- Add one unit test to verify the newly added attribute
- Update Manual Capture documentation

### CODE REVIEW

File: extension/docs/ManualCapture.md [hunk 1]
```diff
@@ -140,6 +141,36 @@ Follow these recommendations when using the idempotency key:
...
+}
+```
+
 ### More info on capture
 
-For more detailed information from Adyen's perspective, see [Adyen's documentation](https://docs.adyen.com/checkout/capture#manual-capture).
+For more detailed information from Adyen's perspective, see following documentation
+
+- [Adyen's documentation](https://docs.adyen.com/checkout/capture#manual-capture).
+- [Adyen refund API](https://docs.adyen.com/api-explorer/#/Payment/v64/post/capture).
```
Comment (lojzatran) [inline extension/docs/ManualCapture.md:176]:
```suggestion
- [Adyen capture API](https://docs.adyen.com/api-explorer/#/Payment/v64/post/capture).
```

File: notification/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "commercetools-adyen-integration-notification",
-  "version": "9.10.0",
+  "version": "9.12.0",
```
Comment (lojzatran) [inline notification/package.json:3]: What about package-lock in notification?

File: extension/src/paymentHandler/manual-capture.handler.js [hunk 3]
```diff
@@ -20,6 +20,8 @@ async function execute(paymentObject) {
       currency: chargeInitialTransaction.amount.currencyCode,
     },
     originalReference: authorizationSuccessTransaction.interactionId,
+    reference:
+      chargeInitialTransaction.custom?.fields?.reference || paymentObject.key,
```
Comment (lojzatran) [inline extension/src/paymentHandler/manual-capture.handler.js:24]: Did it have any reference before? Maybe we should just stick to the default behaviour and not add the payment key as this was not part of this ticket?

**Review by lojzatran — CHANGES_REQUESTED** [review]
Please also test that it works with Adyen and not only mocking.

File: notification/package.json (hunk 2 above)
Comment (leungkinghin) [reply notification/package.json:3]: https://github.com/commercetools/commercetools-adyen-integration/pull/1011/commits/e65a22d5c9b7eb29504ccc6a98587a1be23e3215

File: extension/src/paymentHandler/manual-capture.handler.js (hunk 3 above)
Comment (leungkinghin) [reply extension/src/paymentHandler/manual-capture.handler.js]: https://github.com/commercetools/commercetools-adyen-integration/pull/1011/commits/fc905e630d6ec2b7fe419f037f58ca6b85888b7a

**Comment by leungkinghin** [comment]
> Please also test that it works with Adyen and not only mocking.

I ever removed following code so that real Adyen is made in test case
`scope.post('/capture').reply(200, manualCaptureResponse)`

and the result also contains reference field and test case passed, so i suppose it works with actual Adyen call.
```
[
  {
    action: 'addInterfaceInteraction',
    type: { key: 'ctp-adyen-integration-interaction-payment-type' },
    fields: {
      createdAt: '2022-09-08T22:40:32.693Z',
      response: '{"status":422,"errorCode":"167","message":"Original pspReference required for this operation","errorType":"validation"}',
      request: '{"method":"POST","body":"{\\"modificationAmount\\":{\\"value\\":1000,\\"currency\\":\\"EUR\\"},\\"originalReference\\":\\"8313547924770610\\",\\"reference\\":\\"123456789\\",\\"metadata\\":{},\\"merchantAccount\\":\\"CommercetoolsGmbHDE775\\"}","headers":{"Content-Type":"application/json","X-Api-Key":"***"}}',
      type: 'manualCapture'
    }
  }
]
```

**Comment by lojzatran** [comment]
> > Please also test that it works with Adyen and not only mocking.
> 
> I ever removed following code so that real Adyen is made in test case `scope.post('/capture').reply(200, manualCaptureResponse)`
> 
> and the result also contains reference field and test case passed, so i suppose it works with actual Adyen call.
> 
> ```
> [
>   {
>     action: 'addInterfaceInteraction',
>     type: { key: 'ctp-adyen-integration-interaction-payment-type' },
>     fields: {
>       createdAt: '2022-09-08T22:40:32.693Z',
>       response: '{"status":422,"errorCode":"167","message":"Original pspReference required for this operation","errorType":"validation"}',
>       request: '{"method":"POST","body":"{\\"modificationAmount\\":{\\"value\\":1000,\\"currency\\":\\"EUR\\"},\\"originalReference\\":\\"8313547924770610\\",\\"reference\\":\\"123456789\\",\\"metadata\\":{},\\"merchantAccount\\":\\"CommercetoolsGmbHDE775\\"}","headers":{"Content-Type":"application/json","X-Api-Key":"***"}}',
>       type: 'manualCapture'
>     }
>   }
> ]
> ```

It returns an error here so it is not clear that your change is working. Please try the process again and get a valid response in the end.

**Comment by leungkinghin** [comment]
> > > Please also test that it works with Adyen and not only mocking.
> > 
> > 
> > I ever removed following code so that real Adyen is made in test case `scope.post('/capture').reply(200, manualCaptureResponse)`
> > and the result also contains reference field and test case passed, so i suppose it works with actual Adyen call.
> > ```
> > [
> >   {
> >     action: 'addInterfaceInteraction',
> >     type: { key: 'ctp-adyen-integration-interaction-payment-type' },
> >     fields: {
> >       createdAt: '2022-09-08T22:40:32.693Z',
> >       response: '{"status":422,"errorCode":"167","message":"Original pspReference required for this operation","errorType":"validation"}',
> >       request: '{"method":"POST","body":"{\\"modificationAmount\\":{\\"value\\":1000,\\"currency\\":\\"EUR\\"},\\"originalReference\\":\\"8313547924770610\\",\\"reference\\":\\"123456789\\",\\"metadata\\":{},\\"merchantAccount\\":\\"CommercetoolsGmbHDE775\\"}","headers":{"Content-Type":"application/json","X-Api-Key":"***"}}',
> >       type: 'manualCapture'
> >     }
> >   }
> > ]
> > ```
> 
> It returns an error here so it is not clear that your change is working. Please try the process again and get a valid response in the end.

I tried to make a new payment with ID `9ba1d2ff-ad2e-48f9-8750-1d20b7407936` and captured the pspReference, used it in manual capture, and finally got the following result, in which the requests contains the new reference attribute.
```
[
  {
    action: 'addInterfaceInteraction',
    type: { key: 'ctp-adyen-integration-interaction-payment-type' },
    fields: {
      createdAt: '2022-09-09T10:19:48.650Z',
      response: '{"pspReference":"Z83MNCCR3VTFWR82","response":"[capture-received]"}',
      request: '{"method":"POST","body":"{\\"modificationAmount\\":{\\"value\\":1000,\\"currency\\":\\"EUR\\"},\\"originalReference\\":\\"ZZ9HLDPXNMK2WN82\\",\\"reference\\":\\"123456789\\",\\"metadata\\":{},\\"merchantAccount\\":\\"CommercetoolsGmbHDE775\\"}","headers":{"Content-Type":"application/json","X-Api-Key":"***"}}',
      type: 'manualCapture'
    }
  },
  {
    action: 'changeTransactionState',
    transactionId: 'chargeInitialTransactionId',
    state: 'Pending'
  },
  {
    action: 'changeTransactionInteractionId',
    transactionId: 'chargeInitialTransactionId',
    interactionId: 'Z83MNCCR3VTFWR82'
  }
]

```

**Review by lojzatran — APPROVED** [review]

**Pull request merged**